sqlglot/dialects/tsql.py

```diff
@@ -398,6 +409,22 @@ class Parser(parser.Parser):
 
         CONCAT_NULL_OUTPUTS_STRING = True
 
+        def _parse_set(self, unset: bool = False, tag: bool = False) -> exp.Set | exp.Command:
```

Comment:
this is not correct at all, i think you want to to make set_item_assignment just have optional eq

Reply:
If I understand you correctly, I tried that early on and again after the comment. I was not able to get it to work by manipulating the logic:
```python
        left = self._parse_primary() or self._parse_id_var()

        if not self._match_texts(("=", "TO")):
             self._retreat(index)
             return None

        right = self._parse_statement() or self._parse_id_var()
        this = self.expression(exp.EQ, this=left, expression=right)

        return self.expression(exp.SetItem, this=this, kind=kind)
    ```

Reply:
One desirable outcome, the parsed statement is classed as a SET expression rather than the Command expression

Reply:
I think something along these lines might do the trick:

```diff
+++ b/sqlglot/dialects/tsql.py
@@ -643,6 +643,7 @@ class TSQL(Dialect):
             exp.Min: min_or_least,
             exp.NumberToStr: _format_sql,
             exp.Select: transforms.preprocess([transforms.eliminate_distinct_on]),
+            exp.SetItem: lambda self, e: f"{self.sql(e.this.left)}{' ' if isinstance(e.this.left, exp.Identifier) else ' = '}{self.sql(e.this.right)}",

+++ b/sqlglot/parser.py
@@ -4808,10 +4808,11 @@ class Parser(metaclass=_Parser):

         left = self._parse_primary() or self._parse_id_var()

-        if not self._match_texts(("=", "TO")):
+        if not left:
             self._retreat(index)
             return None

+        self._match_texts(("=", "TO"))
```

Reply:
Hi @GeorgeSittas thanks very much for your helpful input. Let me take a look.